pkg/controllers/proxy/network_services_controller.go

```diff
@@ -394,7 +397,6 @@ func (nsc *NetworkServicesController) Run(healthChan chan<- *healthcheck.Control
 
 		case <-t.C:
 			glog.V(1).Info("Performing periodic sync of ipvs services")
-			healthcheck.SendHeartBeat(healthChan, "NSC")
```

Comment:
This approach makes sense from a minimalistic logic point of view, but I think that it would be a mistake to remove the network services controller heartbeat from the network_services_controller file. I think that putting it in the service_endpoints_sync file makes it harder to find and also it seems like it might get lost there.

My recommendation is that we leave it here, and also place another `SendHeartBeat` call just above the `switch perform` statement on line 381. I think that the intent will be more obvious and it won't be victim of a refactor later.

Reply:
While I agree that code to send explicit heartbeats makes intent more obvious, I am worried that its error prone as one can easily miss to add if there is any future events that can trigger sync. Just like the case this bug fixes it got missed after refactor.

Reply:
Ok, updated the PR as per suggesttion. Effectivley they achive same result, but its just matter of readability, less error prone code.

Plan to raise another PR to better structure health checks and sync's across the controllers.